### DISCUSSION TITLE
How can I find the current network key?

### DISCUSSION THREAD

**Discussion by sergicastellsague** [Koenkk/zigbee2mqtt#11916] [General]
Hey there,

I'm dockerizing my "old" zigbee2mqtt set up, and I cannot find the key the devices have been using 🤯. I would like to find out the current network key to reuse it in my dockerized version, so that I do not have to re-pair all devices (hopefully).

I've checked all files, but without luck. The only mention to the network key is the following file, but it's an empty array.
```
bash-5.1# cat options.json 
{
  "data_path": "/data",
  "homeassistant": true,
  "permit_join": false,
  "mqtt_base_topic": "zigbee2mqtt",
  "mqtt_server": "mqtt://192.168.1.33",
  "serial_port": "/dev/ttyACM0",
  "devices": [],
  "network_key": [],
  "mqtt_user": "hassio",
  "mqtt_pass": "hassio",
  "disable_led": true
}
```

**Comment by aleksarias** [comment]
Bump!

**Comment by Koenkk** [comment]
If no network key is specified, the default key is used: `01:03:05:07:09:0B:0D:0F:00:02:04:06:08:0A:0C:0D`

**Reply by Ceaus** [reply]
Isn't this a huge security risk? Even the docs (*) say it's advised to change it. Out of curiosity, why not generate a random key?

*) https://www.zigbee2mqtt.io/advanced/zigbee/03_secure_network.html#change-zigbee-network-encryption-key

**Comment by aleksarias** [comment]
I’m using a helm chart that automatically generates a key but it doesn’t
log it. Is there any way to view the key via zigbee2mqtt?

Alex

**Reply by Koenkk** [reply]
frontend -> settings -> settings -> advanced.

**Comment by aleksarias** [comment]
Thank you!!!
I saw that and it looks blank. I guess it’s just an empty string.
This is an awesome software project. Thanks for all your hard work!
